pkg/resource/deploy/deployment_executor.go

```diff
@@ -267,8 +275,38 @@ func (ex *deploymentExecutor) Execute(callerCtx context.Context, opts Options, p
 		res = ex.checkTargets(opts.UpdateTargets, OpUpdate)
 	}
 
+	// Check that we did operations for everything expected in the plan. We mutate ResourcePlan.Ops as we run
```

Comment:
Spend a bunch of time reading this.. I think it makes sense given the explanation of how it mutates. Find ResourcePlans with pending ops (shouldn't be any), fail if any of them is not an unneeded delete. Why `res == nil` check?

Reply:
Not really any point checking for plan failures if we're going to fail the deployment anyway. Chances are if the deployment failed a load of expected ops didn't happen so this would just be noise. That should probably be a comment here...